File: backend/src/main/java/com/tripfriend/domain/recruit/recruit/repository/RecruitRepositoryCustom.java
package com.tripfriend.domain.recruit.recruit.repository;

import com.tripfriend.domain.member.member.entity.AgeRange;
import com.tripfriend.domain.member.member.entity.Gender;
import com.tripfriend.domain.recruit.recruit.entity.Recruit;
import com.tripfriend.domain.recruit.recruit.entity.TravelStyle;

import java.time.LocalDate;
import java.util.List;
import java.util.Optional;

public interface RecruitRepositoryCustom {
    List<Recruit> findByRecruitTest();
    List<Recruit> searchByTitleOrContent(String keyword);
    List<Recruit> findByIsClosed(boolean isClosed);
    List<Recruit> searchFilterSort(
            Optional<String> keyword,
            Optional<String> placeCityName,
            Optional<Boolean> isClosed,
            Optional<LocalDate> startDate,
            Optional<LocalDate> endDate,
            Optional<String> travelStyle,
            Optional<Boolean> sameGender,
            Optional<Boolean> sameAge,
            Optional<Integer> minBudget,
            Optional<Integer> maxBudget,
            Optional<Integer> minGroupSize,
            Optional<Integer> maxGroupSize,
            Optional<String> sortBy,
            Gender userGender,
            AgeRange userAgeRange
    );
}

File: backend/src/main/java/com/tripfriend/domain/recruit/recruit/repository/RecruitRepositoryCustom.kt
package com.tripfriend.domain.recruit.recruit.repository

import com.tripfriend.domain.member.member.entity.AgeRange
import com.tripfriend.domain.member.member.entity.Gender
import com.tripfriend.domain.recruit.recruit.entity.Recruit
import java.time.LocalDate

interface RecruitRepositoryCustom {
    fun findByRecruitTest(): List<Recruit>
    fun searchByTitleOrContent(keyword: String): List<Recruit>
    fun findByIsClosed(isClosed: Boolean): List<Recruit>
    fun searchFilterSort(
        keyword: String?,
        placeCityName: String?,
        isClosed: Boolean?,
        startDate: LocalDate?,
        endDate: LocalDate?,
        travelStyle: String?,
        sameGender: Boolean?,
        sameAge: Boolean?,
        minBudget: Int?,
        maxBudget: Int?,
        minGroupSize: Int?,
        maxGroupSize: Int?,
        sortBy: String?,
        userGender: Gender?,
        userAgeRange: AgeRange?
    ): List<Recruit>
}

